tests/crypto_aesctr/main.c

```diff
@@ -241,80 +241,40 @@ perftest(void)
 {
 	struct crypto_aes_key * key_exp;
 	uint8_t key[32];
-	uint8_t * largebuf;
-	struct timeval begin, end;
-	double delta_s;
-	size_t i, j;
-	size_t num_blocks;
-	size_t bufsize;
-	size_t maxbufsize = perfsizes[num_perf - 1];
+	size_t i;
+	PERFTESTS_VARS(0)
 
 	/* Inform user about the hardware optimization status. */
 	print_hardware("Performance test of AES-CTR");
 	fflush(stdout);
 
-	/* Allocate buffer to hold largest message. */
-	if ((largebuf = malloc(maxbufsize)) == NULL) {
-		warnp("malloc");
-		goto err0;
-	}
-	memset(largebuf, 0, maxbufsize);
-
 	/* Prepare the key.  We're only performance-testing 256-bit keys. */
 	for (i = 0; i < 32; i++)
 		key[i] = (uint8_t)i;
 	if ((key_exp = crypto_aes_key_expand(key, 32)) == NULL)
 		goto err1;
 
 	/* Warm up. */
-	for (j = 0; j < 100000; j++)
-		crypto_aesctr_buf(key_exp, j, largebuf, largebuf, perfsizes[0]);
-
-	/* Reset plaintext input. */
-	memset(largebuf, 0, maxbufsize);
+	for (i = 0; i < 100000; i++)
+		crypto_aesctr_buf(key_exp, i, perftests_buf, perftests_buf,
+		    PERFTESTS_MAX_BUFSIZE);
 
 	/* Run operations. */
-	for (i = 0; i < num_perf; i++) {
-		bufsize = perfsizes[i];
-		num_blocks = bytes_to_encrypt / bufsize;
-
-		/* Get beginning time. */
-		if (monoclock_get_cputime(&begin)) {
-			warnp("monoclock_get_cputime()");
-			goto err2;
-		}
-
-		/* Encrypt all the bytes. */
-		for (j = 0; j < num_blocks; j++)
-			crypto_aesctr_buf(key_exp, j, largebuf, largebuf,
-			    bufsize);
-
-		/* Get ending time. */
-		if (monoclock_get_cputime(&end)) {
-			warnp("monoclock_get_cputime()");
-			goto err2;
-		}
-
-		/* Prepare output. */
-		delta_s = timeval_diff(begin, end);
-
-		/* Print results. */
-		printf("... in %zu blocks of size %zu:\t%.06f s\t%.01f MB/s\n",
-		    num_blocks, bufsize, delta_s,
-		    (double)bytes_to_encrypt / 1e6 / delta_s);
-		fflush(stdout);
-	}
-
-	/* Clean up. */
-	free(largebuf);
+	PERFTESTS_LOOP_OUTER_BEGIN
```

Comment:
I considered calling this something like `PERFTESTS_BUFSIZE_CHOOSE_BEGIN` (although not that precisely!), but I haven't come up with anything as clear as "begin outer loop".

Another nitpick might be the word order.  I went with "loop outer begin" so that if one were to sort the macros of `perftests.h` alphabetically, related functions would appear next to each other.  (I picked up that trick from naming lilypond commands, thinking of the function name index.)

That said, I cannot think of a reason one would want to sort the macros alphabetically.  If we had a dozens of them in the same file, there might be some scenario in which that would be useful.